### ISSUE TITLE
React Native 0.7

### ISSUE THREAD

**Issue by grabbou** [mjohnston/react-native-webpack-server#40]
Not sure if you guys have issues as well, but I keep getting:

```
Server listening at http://localhost:8080
Webpack dev server listening at  http://localhost:8082
Unhandled rejection NotFoundError: Cannot find entry file index.ios.js in any of the roots: ["/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native","/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native-webpack-server/_entry"]
    at DependecyGraph.getOrderedDependencies (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/DependencyGraph/index.js:95:11)
    at /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/index.js:96:35
    at tryCatcher (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/promise.js:489:31)
    at Promise._settlePromiseAt (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/promise.js:565:18)
    at Async._drainQueue (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:358:17)
Unhandled rejection SyntaxError: TransformError: /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js: ENOENT, open '/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js'
[BABEL] The transformer es6.parameters.rest has been renamed to es6.parameters
[17:17:44] <START> find dependencies
[17:17:44] <END>   find dependencies (21ms)
[17:17:44] <START> transform
Hash: ea23e60d7042c5694560
Version: webpack 1.9.12
Time: 6395ms
           Asset     Size  Chunks             Chunk Names
    index.ios.js  1.56 MB       0  [emitted]  index.ios
index.ios.js.map  1.79 MB       0  [emitted]  index.ios
chunk    {0} index.ios.js, index.ios.js.map (index.ios) 1.5 MB [rendered]
webpack: bundle is now VALID.
Unhandled rejection SyntaxError: TransformError: /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js: ENOENT, open '/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js'
Unhandled rejection SyntaxError: TransformError: /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js: ENOENT, open '/Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native/packager/react-packager/src/DependencyResolver/haste/polyfills/prelude_dev.js'
[BABEL] The transformer es6.parameters.rest has been renamed to es6.parameters
[17:17:51] <START> find dependencies
[17:17:51] <END>   find dependencies (35ms)
[17:17:51] <START> transform

 ===============================================================
 |  Running packager on port 8081.       
 |  Keep this packager running while developing on any JS         
 |  projects. Feel free to close this tab and run your own      
 |  packager instance if you prefer.                              
 |                                                              
 |     https://github.com/facebook/react-native                 
 |                                                              
 ===============================================================

Looking for JS files in
   /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native
   /Users/grabbou/Repositories/ExplosiveTitles/este-native/node_modules/react-native-webpack-server/_entry 

 ERROR  Packager can't listen on port 8081
Most likely another process is already using this port
Run the following command to find out which process:

   lsof -n -i4TCP:8081 

You can either shut down the other process:

   kill -9 <PID> 

or run packager on different port.

See http://facebook.github.io/react-native/docs/troubleshooting.html
for common problems and solutions.
```

**Comment by leoasis** [comment]
I'm having a different set of errors but also only happening on 0.7:

```
Error: File /Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native-collapsible/Accordion.js not found in any of the roots
    at Fastfs._getAndAssertRoot (/Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/packager/react-packager/src/DependencyResolver/fastfs.js:137:13)
    at Fastfs._add (/Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/packager/react-packager/src/DependencyResolver/fastfs.js:152:10)
    at /Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/packager/react-packager/src/DependencyResolver/fastfs.js:38:18
    at Array.forEach (native)
    at /Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/packager/react-packager/src/DependencyResolver/fastfs.js:31:13
    at tryCallOne (/Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/node_modules/promise/lib/core.js:37:12)
    at /Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/node_modules/promise/lib/core.js:103:15
    at flush (/Users/lenny/dev/toptal/power_supply/nosh-ios/node_modules/react-native/node_modules/promise/node_modules/asap/raw.js:50:29)
    at process._tickCallback (node.js:355:11)
{"type":"InternalError","message":"react-packager has encountered an internal error, please check your terminal error output for more details"}
[5:57:45 PM] <START> request:/index.ios.bundle
[5:57:45 PM] <START> find dependencies
```

It's not related only to `react-native-collapsible`, since if I don't use it, that error prompts with some other file. And actually, I'm never using the `Accordion.js` file, so it shouldn't even look for it

**Comment by grabbou** [comment]
Yeah, looks like somehow all folders from  `node_modules` are taken if they match `react-native-*` name

**Comment by leoasis** [comment]
I think it's related to this issue https://github.com/facebook/react-native/issues/1437

**Comment by grabbou** [comment]
Yeah, I've been experiencing that issue as well. I know React Native is still in early stage, but the number of BC or regressions is just driving me crazy as we are about to launch the app (at least 0.6 works super smoothly!)

**Comment by ptmt** [comment]
Keep getting errors like `Error: File /Users/potomushto/Projects/moneyed/node_modules/react-native-vector-icons/node_modules/yargs/node_modules/cliui/node_modules/right-align/node_modules/align-text/node_modules/repeat-string/index.js not found in any of the roots`

**Comment by ptmt** [comment]
Updated `Server.js`, works for me. I need 0.8.0 and I love hot reloading, so had no choice. Would be awesome to learn webpack-server more deeply and contribute the right solution.

**Comment by RyGuyM** [comment]
@unknownexception What was your update to `Server.js` that fixed it for you? I'm getting the same react-native-vector-icons error.

**Comment by ptmt** [comment]
@RyGuyM see the pull request above message . It seems the packager has been rewritten with fastfs. Need to figure out how to pass correct paths.

**Comment by jeffreywescott** [comment]
My pull request #43 should make things work on 0.8.0-rc.2.

**Comment by AbrahamAlcaina** [comment]
@jeffreywescott thanks it is working fine for me. 
:+1:

**Comment by elliottsj** [comment]
Closed via #43
